src/pages/Lobby.js

```diff
@@ -22,178 +18,299 @@ import SaveAlt from '@material-ui/icons/SaveAlt';
-                            color: '#333333'
-                        },
-                        paging: false,
-                        searchFieldStyle:{
-                            fontSize: "14px",
-                            fontFamily: 'verdana'
-                        }
-                    }}
-                    localization = {{
-                        toolbar: {
-                            searchPlaceholder: "keywords"
-                        }
-                    }}
-                />
-                </div>
-            </Container>
-        )
+      },
+    });
+
+  }
+
+  componentWillUnmount(){
+    this.gameUpdateSubscription.unsubscribe()
+    this.gameCreationSubscription.unsubscribe()
+    //this.gameDeletionSubscription.unsubscribe()
+  }
+
+  createGame = async (event, gameInfo) => {
+    this.setState({showDialog: false})
+    gameInfo['fen'] = "init"
+    gameInfo['available'] = true
+    let userInfo = await Auth.currentUserInfo()
+    if(userInfo) {
+      let user = {}
+      user.id = userInfo.id
+      user.username = userInfo.username
+      gameInfo['creator'] = user
     }
+    let currentGame = localStorage.getItem(CURRENT_GAME)
+    if(currentGame){
+       API.graphql(graphqlOperation(mutations.deleteGame, { input: {id: currentGame }}))
+    }
+    let newGame = await API.graphql(graphqlOperation(mutations.createGame, { input: gameInfo }))
+    localStorage.setItem(CURRENT_GAME, newGame.data.createGame.id)
+  }
+
+  showDialog = () => {
+    this.setState({showDialog: true})
+  }
+
+  closeDialog = () => {
+    this.setState({showDialog: false})
+  }
+
+  showJoiningOwnGameDialog = () => {
+    this.setState({showJoiningOwnGameDialog: true})
+  }
+
+  closeJoiningOwnGameDialog = () => {
+    this.setState({showJoiningOwnGameDialog: false})
+  }
+
+  constructRowFromGameData = (game) => {
+    const creator = game.creator
+    const player = creator ? creator.username : 'anonymous'
+    const skillLevel = 'n/a'
+    const timing = game.time;
+    const variant = game.variant
+    const gameId = game.id
+    return {player, skillLevel, timing, variant, gameId}
+  }
+
+  joinGame = async (event, rowData) => {
+    let gameId = rowData.gameId
+    let createdGame = localStorage.getItem(CURRENT_GAME)
+    if(createdGame !== null && createdGame === gameId){
+      this.showJoiningOwnGameDialog()
+      return
+    }
+    let gameInfo = this.gamesData[gameId]
+    gameInfo['available'] = false
+    let userInfo = await Auth.currentUserInfo()
+    if(userInfo) {
+      let opponent = {}
+      opponent.id = userInfo.id
+      opponent.username = userInfo.username
+      gameInfo['opponent'] = opponent
+    }
+    delete gameInfo['__typename']
+    let creator = gameInfo['creator']
+    if(creator) delete creator['__typename']
+    let opponent = gameInfo['opponent']
+    if(opponent) delete opponent['__typename']
+    gameInfo['creator'] = creator
+    gameInfo['opponent'] = opponent
```

Comment:
gameInfo['opponent'] are assigned multiple times here, this is confusing

Reply:
appsync appends a '__typename' to all queries. If that fields is not removed, mutations will no go through. Because Opponent is User, a '__typename' field is attached to it. Another one is also attached on gameInfo. I had to remove them both.